test/vtep/vxlan-responder.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+"""
+SPDX-License-Identifier: Apache-2.0
+Copyright 2021 Authors of Cilium
+
+# vxlan-responder.py -h
+usage: vxlan-responder.py [-h] [--dport UDPPORT] [--sport SRCPORT] [--vni VXLANVNI] [--outerip DSTHOST]
+                          [--innerip INNERHOST] [--bridge BRIDGE]
+
+optional arguments:
+  -h, --help           show this help message and exit
+  --dport UDPPORT      VXLAN UDP port
+  --sport SRCPORT      UDP source port
+  --vni VXLANVNI       VXLAN VNI value
+  --outerip DSTHOST    Outer dst IP
+  --innerip INNERHOST  Inner dst IP
+  --bridge BRIDGE      Bridge interface to sniff
+"""
+
+import argparse
+from scapy import all
+from scapy.layers import all
+from scapy.layers.inet import IP, ICMP, UDP
+from scapy.packet import ls, Raw
+from scapy.sendrecv import sniff, send
+from scapy.all import *
+PAYLOAD='zzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzz'
+
+
+def udp_monitor_callback(pkt):
+    inLayer3 = pkt.payload.payload.payload.payload.payload
+
+    if(pkt.haslayer(IP) and inLayer3.dst == INNERIP):
+        print("incoming IP packet matches",  INNERIP)
+        outLayer3 = pkt.payload
+        udpLayer = pkt.payload.payload
+        vxlanLayer = pkt.payload.payload.payload
+        inLayer2 = pkt.payload.payload.payload.payload
+        inLayer4 = pkt.payload.payload.payload.payload.payload.payload
+
+        outerIP=IP(src=outLayer3.dst, dst=outLayer3.src)
+
+        udpinfo=UDP(sport=SRCPORT, dport=UDPPORT)
+
+        vxlan=VXLAN(flags=vxlanLayer.flags, vni=VNI)
+
+        innerETH=Ether(dst=inLayer2.src, src=inLayer2.dst, type=0x800)
+
+        innerIP=IP(src=inLayer3.dst,dst=inLayer3.src)
+
+        innerICMP=ICMP(type=0, code=0, id=inLayer4.id, seq=inLayer4.seq)
```

Comment:
nit: I suspect something is a little off with this code as I see ping output from my terminal like this:

```
$ kubectl exec -it busybox-master  -- ping -c 10 10.1.5.1
PING 10.1.5.1 (10.1.5.1): 56 data bytes
64 bytes from 10.1.5.1: seq=0 ttl=63 time=1387047.865 ms
64 bytes from 10.1.5.1: seq=1 ttl=63 time=1387996.872 ms
64 bytes from 10.1.5.1: seq=2 ttl=63 time=1389009.280 ms
64 bytes from 10.1.5.1: seq=3 ttl=63 time=1390004.674 ms
64 bytes from 10.1.5.1: seq=4 ttl=63 time=1391013.255 ms
64 bytes from 10.1.5.1: seq=5 ttl=63 time=1392000.446 ms
64 bytes from 10.1.5.1: seq=6 ttl=63 time=1393009.569 ms
64 bytes from 10.1.5.1: seq=7 ttl=63 time=1393997.417 ms
64 bytes from 10.1.5.1: seq=8 ttl=63 time=1395004.283 ms
64 bytes from 10.1.5.1: seq=9 ttl=63 time=1396020.220 ms

--- 10.1.5.1 ping statistics ---
10 packets transmitted, 10 packets received, 0% packet loss
round-trip min/avg/max = 1387047.865/1391510.388/1396020.220 ms
```

However Linux is calculating these round-trip latencies, it thinks that a lot more time has passed than the actual RTT.

That said, this is not a functional issue with the feature, it's just strange to observe when trying the demo. Would be a "nice to fix" but not a requirement for merging.

Reply:
this may take me some time to investigate, I will leave it for now after I addressed the remaining major issue of this feature like supporting network policy...etc.